smtp.c

```diff
@@ -6,6 +6,9 @@
 #include <time.h>
 #ifdef HAVE_GETRANDOM
 #include <sys/random.h>
+#elif defined HAVE_GETRANDOM_SYSCALL
```

Comment:
How about putting `#define _GNU_SOURCE` inside this conditional block, rather than adding it to the CFLAGS? The CFLAGS affects all source files, and it seems we really only need it here.

Reply:
That was my initial idea, but I think `_GNU_SOURCE` has to be defined before any includes. Unless it is, the compiler complains that `syscall` is implicitly declared. Hence the inline definition, which avoids that.

Reply:
Interesting, I thought that if `_GNU_SOURCE` was defined before the inclusion of `unistd.h` then that would be enough...

Did you test putting it there, and found the compiler complained? If so, how about adding this to the top of smtp.c instead:

```c
#ifdef HAVE_GETRANDOM_SYSCALL
#define _GNU_SOURCE
#endif
```

I'd like to isolate _GNU_SOURCE to exactly the file where it's needed, because it's rather [aggressive](https://www.gnu.org/software/libc/manual/html_node/Feature-Test-Macros.html#index-_005fGNU_005fSOURCE).

Also which distro and version are you using? I'd like to test the build process as well.

Reply:
Looks like having two separate `#ifdef HAVE_GETRANDOM_SYSCALL` checks (with one at the very top of `smtp.c`) could do it. The slight downside is that `_GNU_SOURCE` will be defined even if the `getrandom()` function exists, but I don't know how important that is.

Reply:
Actually it might not, because if HAVE_GETRANDOM then the `elif` won't happen in configure. Looks like you made HAVE_GETRANDOM and HAVE_GETRANDOM_SYSCALL mutually exclusive, which is nice.